README.md

```diff
@@ -10,125 +10,43 @@ $ npm install p-graph
-funcs.set("putOnShorts", () => Promise.resolve("put on your shorts"));
-funcs.set("putOnJacket", () => Promise.resolve("put on your jacket"));
-funcs.set("putOnShoes", () => Promise.resolve("put on your shoes"));
-funcs.set("tieShoes", () => Promise.resolve("tie your shoes"));
-
-const depMap = new Map();
-
-depMap.set(tieShoes, new Set([putOnShoes]));
-depMap.set(putOnJacket, new Set([putOnShirt, putOnShorts]));
-depMap.set(putOnShoes, new Set([putOnShorts]));
-depMap.set(putOnShorts, new Set());
-depMap.set(putOnShirt, new Set());
-
-await pGraph(namedFunctions, graph);
-```
-
-### Using the ID as argument to the same function
-
-In many cases, the jobs that need to run are the same where the only difference is the arguments for the function. In that case, you can treat the IDs as arguments as they are passed into the Run Function.
-
-```ts
-type Id = unknown;
-type RunFunction = (id: Id) => Promise<unknown>;
-```
-
-As you can see, the ID can be anything. It will be passed as the argument for your Run Function. This is a good option if having a large number of functions inside a graph is prohibitive in memory sensitive situations.
+There are some contexts where you may want to limit the number of functions running concurrently. One example would be to prevent overloading the CPU with too many parallel tasks. The concurrency argument to `run` will limit the number of functions that start running at a given time
 
 ```js
-const funcs = new Map();
-const thatImportantTask = (id) => Promise.resolve(id);
-
-funcs.set("putOnShirt", thatImportantTask);
-funcs.set("putOnShorts", thatImportantTask);
-funcs.set("putOnJacket", thatImportantTask);
-funcs.set("putOnShoes", thatImportantTask);
-funcs.set("tieShoes", thatImportantTask);
+await pGraph(graph).run({ maxConcurrency: 3 });
 ```
 
-## Scopes and filtering
+## Priority
 
-After a graph are sent to the `pGraph` function, the graph is executed with the `run()` function. The `run()` takes in an argument that lets you filter which tasks to end. This allows you to run tasks up to a certain point in the graph.
-
-```js
-// graph is one of the three options up top
-// depMap is the dependency map where the key is the ID for the Run Function
-//   - the ID CAN be the Run Function itself if graph is specified as the dependency array format
-await pGraph(graph).run((depMap) => {
-  return [...depMap.keys()].filter((id) => id.startsWith("b"));
-});
-```
+When the concurrency option is set, the task runner must pick a subset of available tasks to put on the queue. By default, tasks are considered to all be equally important and equally likely to be picked to run once all the tasks they depend on are complete. If you wish to control the ordering of tasks, consider using the priority option when defining a task node. When the task scheduler is picking tasks to run, it will favor tasks with a higher priority over tasks with a lower priority. Tasks will always execute in dependency order.
```

Comment:
I think priority and concurrency are separate concepts. Priority merely determines order while concurrency is to help take advantage of multi-core architectures.

Reply:
Well priority only has a discernable impact when you have a limit on concurrency, otherwise we simply run all tasks as soon as they get unblocked. Will reword

Reply:
I said the same thing without using concurrency to avoid confusion